kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+class FallbackLinearSolver
+    : public LinearSolver<TSparseSpaceType, TDenseSpaceType, TReordererType> {
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of LinearSolver
+    KRATOS_CLASS_POINTER_DEFINITION(FallbackLinearSolver);
+
+    /// Definition of the base class
+    using BaseType = LinearSolver<TSparseSpaceType, TDenseSpaceType, TReordererType>;
+
+    /// Definition of the base type pointer
+    using LinearSolverPointer = typename BaseType::Pointer;
+
+    /// The definition of the linear solver factory
+    using LinearSolverFactoryType = LinearSolverFactory<TSparseSpaceType, TDenseSpaceType>;
+
+    /// Type definition for sparse matrix
+    using SparseMatrixType = typename BaseType::SparseMatrixType;
+
+    /// Type definition for pointer to sparse matrix
+    using SparseMatrixPointerType = typename BaseType::SparseMatrixPointerType;
+
+    /// Type definition for vector
+    using VectorType = typename BaseType::VectorType;
+
+    /// Type definition for pointer to vector
+    using VectorPointerType = typename BaseType::VectorPointerType;
+
+    /// Type definition for dense matrix
+    using DenseMatrixType = typename BaseType::DenseMatrixType;
+
+    /// Type definition for dense vector
+    using DenseVectorType = typename BaseType::DenseVectorType;
+
+    /// Type definition for size
+    using SizeType = std::size_t;
+
+    /// Type definition for index
+    using IndexType = std::size_t;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief This is the default constructor
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(Parameters ThisParameters = Parameters(R"({})"))
+        : mParameters(ThisParameters)
+    {
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Define the solvers
+        const SizeType number_of_solver_parameters = mParameters["solvers"].size();
+        KRATOS_ERROR_IF(number_of_solver_parameters == 0) << "No solvers defined in the parameters" << std::endl;
+        mSolvers.reserve(number_of_solver_parameters);
+        for (IndexType i = 0; i < number_of_solver_parameters; ++i) {
+            mSolvers.push_back(ConstructLinearSolverFromSettings(mParameters["solvers"][i]));
+        }
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /**
+     * @brief Constructor with two solvers
+     * @param pSolver1 The first LinearSolverPointer to set.
+     * @param pSolver2 The second LinearSolverPointer to set.
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(
+        LinearSolverPointer pSolver1,
+        LinearSolverPointer pSolver2,
+        Parameters ThisParameters = Parameters(R"({})")
+        ) : mSolvers({pSolver1, pSolver2}),
+            mParameters(ThisParameters)
```

Comment:
this seems like a redundant constructor (there's already one taking an entire `vector` of solvers)

Reply:
May be, but this is easier to use and in mnay cases 2 solvers is enough

Reply:
I agree with @matekelemen , the usage is pretty simple (only requires to add one `{}`), which IMO does not justify another CTor

Reply:
I was about to point the very same. I agree with @matekelemen and @philbucher on this.

Indeed, and thinking in the long run, I'd like to only have the parameters-based contructor (but I can understand that this can be maybe too restrictive).

Reply:
Okay, I will remove it